go-controller/pkg/ovn/master.go

```diff
@@ -1122,23 +1122,36 @@ func (oc *Controller) clearInitialNodeNetworkUnavailableCondition(origNode, newN
 }
 
 // delete chassis of the given nodeName/chassisName map
+// from chassis & chassis_private table
 func deleteChassis(ovnSBClient goovn.Client, chassisMap map[string]string) {
 	cmds := make([]*goovn.OvnCommand, 0, len(chassisMap))
 	for chassisHostname, chassisName := range chassisMap {
 		if chassisName != "" {
 			klog.Infof("Deleting stale chassis %s (%s)", chassisHostname, chassisName)
-			cmd, err := ovnSBClient.ChassisDel(chassisName)
+			chassisDelCmd, err := ovnSBClient.ChassisDel(chassisName)
 			if err != nil {
 				klog.Errorf("Unable to create the ChassisDel command for chassis: %s from the sbdb", chassisName)
 			} else {
-				cmds = append(cmds, cmd)
+				cmds = append(cmds, chassisDelCmd)
+			}
+			// check for chassis_private table in schema and
+			// if present, delete corresponding chassis row in chassis_private table
+			sbDbSchema := ovnSBClient.GetSchema()
+			if _, ok := sbDbSchema.Tables[goovn.TableChassisPrivate]; ok {
+				chassisPrivateDelCmd, err := ovnSBClient.ChassisPrivateDel(chassisName)
+				if err != nil {
+					klog.Errorf("Unable to create the ChassisPrivateDel command for chassis: %s from the sbdb", chassisName)
+				} else {
+					cmds = append(cmds, chassisPrivateDelCmd)
+				}
 			}
 		}
 	}
 
 	if len(cmds) != 0 {
 		if err := ovnSBClient.Execute(cmds...); err != nil {
-			klog.Errorf("Failed to delete chassis for node/chassis map %v: error: %v", chassisMap, err)
+			klog.Errorf("Failed to delete chassis row from chassis & chassis_private table "+
```

Comment:
well, it could be that the OVN doesn't have chassis_private table, but we would end up printing this message.. it's a NIT